File: leetcode/Remove_Duplicates_from_Sorted_Array.java
import java.util.*;

public class Solution {
    public int removeDuplicates(int[] A) {
        if (A.length == 0) return 0;
        int cur = 0;
        for (int i = 1; i < A.length; ++i) {
            if (A[cur] != A[i]) {
                ++cur;
                A[cur] = A[i];
            }
        }
        return cur + 1;
    }
}

File: leetcode/Remove_Duplicates_from_Sorted_Array.cpp
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        int new_n = 0;
        for (int i = 0; i < n; ++i)
            if (i == 0 || A[i] != A[i - 1])
                A[new_n++] = A[i];
        return new_n;
    }
};
